apis/v1/gateway_types.go

```diff
@@ -440,6 +449,44 @@ type GatewayTLSConfig struct {
 	Options map[AnnotationKey]AnnotationValue `json:"options,omitempty"`
 }
 
+// ClientValidationContext holds configuration that can be used to validate the client intiating the TLS connection
+// to the Gateway.
+// By default, no client specific configuration is validated.
+type ClientValidationContext struct {
+	// CACertificateRefs contains one or more references to
+	// Kubernetes objects that contain TLS certificates of
+	// the Certificate Authorities that can be used
+	// as a trust anchor to validate the certificates presented by the client.
+	//
+	// A single CA certificate reference to a Kubernetes ConfigMap kind has "Core" support.
+	// Implementations MAY choose to support attaching multiple CA certificates to
+	// a Listener, but this behavior is implementation-specific.
+	//
+	// Support: Core - An optional single reference to a Kubernetes Secret and ConfigMap,
+	// with the CA certificate in a key named `ca.crt`.
+	//
+	// Support: Implementation-specific (More than one reference, or other kinds
+	// of resources).
+	//
+	// References to a resource in a different namespace are invalid UNLESS there
+	// is a ReferenceGrant in the target namespace that allows the certificate
+	// to be attached. If a ReferenceGrant does not allow this reference, the
+	// "ResolvedRefs" condition MUST be set to False for this listener with the
+	// "RefNotPermitted" reason.
+	//
+	// +kubebuilder:validation:MaxItems=8
+	// +kubebuilder:validation:MinItems=1
+	CACertificateRefs []SecretObjectReference `json:"caCertificateRefs"`
+
+	// SubjectAltNames contains one or more alternate names to verify
+	// the subject identity in the certificate presented by the client.
+	//
+	// Support: Core
+	//
+	// +optional
+	SubjectAltNames []string `json:"subjectAltNames,omitempty"`
```

Comment:
nit: min=1,max=??

Reply:
Good question about max? I think maybe we should leave it unspecified (it's implementation defined in X.509)?

cc @arkodg, implementation defined or not, I think this needs some update(s) in #2273

Reply:
This field has been removed altogether in the latest version of the GEP.